src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/record-handled-exceptions.mdx

```diff
@@ -0,0 +1,387 @@
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.recordHandledException(Exception $exceptionToHandle)
+NewRelic.recordHandledException(Exception $exceptionToHandle, Map of String, Object $exceptionAttributes)
+```
+
+```java
+NewRelic.recordHandledException(Throwable $throwableToHandle)
+NewRelic.recordHandledException(Throwable $throwableToHandle, Map of String, Object $exceptionAttributes)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+  NewRelic.recordHandledException(exception: Exception?)
+```
+
+```kotlin
+
+```
+
+## Description [#description]
+
+Records a handled exception or other throwable type. Optionally takes map with additional attributes showing context.
+
+Use `recordHandledException()` within a `try{...} catch(){...}` block to help understand how often your application is throwing exceptions, and under what conditions.
+
+In addition to associated custom attributes, the events will also have associated [session attributes](/docs/insights/insights-data-sources/default-attributes/mobile-default-attributes-insights). You can view event data in the mobile monitoring UI in the [Crash event trail](/docs/mobile-monitoring/mobile-monitoring-ui/crashes/mobile-crash-event-trail), or via NRQL.
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width="30%">
+        Parameter
+      </th>
+
+      <th width="30%">
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$exceptionToHandle`
+      </td>
+
+      <td>
+        `Exception`
+      </td> 
+
+      <td>
+       Required. The exception to be recorded.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$exceptionAttributes`
+      </td>
+
+      <td>
+        `Map of String, Object`
+      </td>
+      <td>
+        Optional. A map of attributes to be associated with the exception.
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if recorded successfully, or `false` if not.
+
+## Examples [#examples]
+
+Here's an example of recording a `ClassCastException` from within an on-click listener:
+
+### Java [#java]
+
+```java
+public class MainActivity extends Activity {
+  ...
+  coolButton.setOnClickListener(new View.OnClickListener() {
+            Map myMap = new HashMap<>();
+            @Override
+            public void onClick(View view) {
+                try {
+                    myMap.put("Key", "Value");
+                    Integer stringVar = (Integer) myMap.get("Key"); //throws ClassCastException
+                } catch (Exception e) {
+                    NewRelic.recordHandledException(e, myMap);
+                }
+            }
+        });
+  ...
+}
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
class MainActivity : AppCompatActivity() {

    ....
        binding.fab.setOnClickListener { view ->
            val myMap = mutableMapOf<String,Any>()
            try {
                myMap["Key"] = "Value"
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAnchorView(R.id.fab)
                    .setAction("Action", null).show()
            }catch (exception:Exception) {
                NewRelic.recordHandledException(exception,myMap)
            }
        }
    ........
    }

Reply:
```suggestion
```kotlin

class MainActivity : AppCompatActivity() {

....
    binding.fab.setOnClickListener { view ->
        val myMap = mutableMapOf<String,Any>()
        try {
            myMap["Key"] = "Value"
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                .setAnchorView(R.id.fab)
                .setAction("Action", null).show()
        }catch (exception:Exception) {
            NewRelic.recordHandledException(exception,myMap)
        }
    }
........

```
```